build_prod.py

```diff
@@ -210,9 +253,32 @@ def add_meta_layers(cfg):
             yocto_add_bblayer(cfg, bblayer)
 
 
-def build_init(uri, branch, xml_base_name):
+def build_init(uri, branch, xml_base_name, proposed):
+    if not branch:
+        branch = 'master'
+
     repo_init(uri, branch, xml_base_name)
     repo_sync()
+    c_dir = os.getcwd()
+    # go into prod layer directory to get the repo instance 
+    # and create the remote.
+    # (It is required to apply the pull request).
+    os.chdir('meta-xt-{}'.format(xml_base_name))
+    # Get url of prod-layer to find the pull requests
+    # The name of repository is known but organization - not
+    # The organization can be extracted from theurl of layer 'meta-xt-products'
+    # Assumption:the layers  products and prod belong 
+    # to the same organization    
+    url = re.search('https://github.com/(.+?)products.git', uri).group(1)
+    url += xml_base_name
+    repo = git.Repo('./')
+    repo.create_remote('origin', url='git://github.com/{}'.format(url))
```

Comment:
Let me suggest to change name of remote from 'origin' to something not so widely used. For example 'pull_req_source'.
Reason for that change is that we have distributed configuration across few projects, and we specify name of remote in product manifest in meta-xt-products. It will be strange to specify to not use 'origin' as default remote name because it may overlap with our build script. And there is no guarantee that some one will set default name of remote to 'origin'. So it's better to avoid usage of such common name.

Reply:
changed